txpipe/maps.py

```diff
@@ -0,0 +1,503 @@
+            # keys are the output tag and the map name
+            maps["source_maps", f"g1_{b}"] = (pix, g1[b])
+            maps["source_maps", f"g2_{b}"] = (pix, g2[b])
+            maps["source_maps", f"var_g1_{b}"] = (pix, var_g1[b])
+            maps["source_maps", f"var_g1_{b}"] = (pix, var_g1[b])
+            maps["source_maps", f"lensing_weight_{b}"] = (pix, var_g1[b])
+
+        return maps
+
+
+class TXLensMaps(TXBaseMaps):
+    """
+    Make galaxy number count maps from photometry
+    and lens tomography.
+
+    Density maps are made later once masks are generated.
+    """
+
+    name = "TXLensMaps"
+
+    inputs = [
+        ("photometry_catalog", HDFFile),
+        ("lens_tomography_catalog", TomographyCatalog),
+    ]
+
+    outputs = [
+        ("lens_maps", MapsFile),
+    ]
+
+    config_options = {**map_config_options}
+
+    def prepare_mappers(self, pixel_scheme):
+        # read nbin_lens and save
+        with self.open_input("lens_tomography_catalog") as f:
+            nbin_lens = f["tomography"].attrs["nbin_lens"]
+        self.config["nbin_lens"] = nbin_lens
+
+        # create lone mapper
+        lens_bins = list(range(nbin_lens))
+        source_bins = []
+        mapper = Mapper(
+            pixel_scheme,
+            lens_bins,
+            source_bins,
+            do_g=False,
+            sparse=self.config["sparse"],
+        )
+        return [mapper]
+
+    def data_iterator(self):
+        print("TODO: no lens weights here")
+        # see TXSourceMaps abov for info on this
+        return self.combined_iterators(
+            self.config["chunk_rows"],
+            # first file
+            "photometry_catalog",
+            "photometry",
+            ["ra", "dec"],
+            # next file
+            "lens_tomography_catalog",
+            "tomography",
+            ["lens_bin"],
+        )
+
+    def accumulate_maps(self, pixel_scheme, data, mappers):
+        # no need to rename cols here since just ra, dec, lens_bin
+        mapper = mappers[0]
+        mapper.add_data(data)
+
+    def finalize_mappers(self, pixel_scheme, mappers):
+        # Again just the one mapper
+        mapper = mappers[0]
+        # Ignored return values are empty dicts for shear
+        pix, ngal, _, _, _, _, _ = mapper.finalize(self.comm)
+        maps = {}
+
+        if self.rank != 0:
+            return maps
+
+        for b in mapper.lens_bins:
+            # keys are the output tag and the map name
+            maps["lens_maps", f"ngal_{b}"] = (pix, ngal[b])
+
+        return maps
+
+
+class TXExternalLensMaps(TXLensMaps):
+    """
+    Same as TXLensMaps except it reads from an external
+    lens catalog.
+    """
+
+    name = "TXExternalLensMaps"
+
+    inputs = [
+        ("lens_catalog", HDFFile),
+        ("lens_tomography_catalog", TomographyCatalog),
+    ]
+
+    config_options = {**map_config_options}
+
+    def data_iterator(self):
+        print("TODO: no lens weights here")
```

Comment:
why no weights here? or do you mean you're gonna add it later?

Reply:
Yes, meant it was to add later.  I've updated for clarity